tests/server/wildfly/src/test/java/org/hawkular/apm/tests/wildfly/zipkin/SpanAnalyticsITest.java

```diff
@@ -0,0 +1,253 @@
+                        SpanServiceITest.millisToMicroSeconds(1), SpanServiceITest.millisToMicroSeconds(30)));
+        rootSpan.setTimestamp(SpanServiceITest.millisToMicroSeconds(1L));
+        rootSpan.setDuration(SpanServiceITest.calculateDuration(1L, 30L));
+        rootSpan.setTraceId("root");
+        rootSpan.setId("root");
+        rootSpan.setName("get");
+
+        BinaryAnnotation urlServiceABinAnn = new BinaryAnnotation();
+        urlServiceABinAnn.setKey(Constants.ZIPKIN_BIN_ANNOTATION_HTTP_URL);
+        urlServiceABinAnn.setValue("http://localhost/serviceA");
+
+        Span clientSpan = new Span(Arrays.asList(urlServiceABinAnn),
+                SpanServiceITest.clientAnnotation(SpanServiceITest.millisToMicroSeconds(5),
+                        SpanServiceITest.millisToMicroSeconds(15)));
+        clientSpan.setTimestamp(SpanServiceITest.millisToMicroSeconds(5));
+        clientSpan.setDuration(SpanServiceITest.calculateDuration(5L, 15L));
+        clientSpan.setTraceId(rootSpan.getTraceId());
+        clientSpan.setId("2");
+        clientSpan.setParentId("root");
+        clientSpan.setName("post");
+
+        Span serverSpan = new Span(Arrays.asList(urlServiceABinAnn),
+                SpanServiceITest.serverAnnotations(
+                        SpanServiceITest.millisToMicroSeconds(7), SpanServiceITest.millisToMicroSeconds(10)));
+        serverSpan.setTimestamp(SpanServiceITest.millisToMicroSeconds(7));
+        serverSpan.setDuration(SpanServiceITest.calculateDuration(7L, 10L));
+        serverSpan.setTraceId(rootSpan.getTraceId());
+        serverSpan.setId("2");
+        serverSpan.setParentId("root");
+        serverSpan.setName("post");
+
+        post(Server.Zipkin, "/spans", null, Arrays.asList(rootSpan));
+        post(Server.Zipkin, "/spans", null, Arrays.asList(clientSpan));
+        post(Server.Zipkin, "/spans", null, Arrays.asList(serverSpan));
+
+        Wait.until(() -> {
+            Collection<NodeSummaryStatistics> nodeSummaryStatistics =
+                    analyticsService.getNodeSummaryStatistics(null, requestCriteria());
+            return nodeSummaryStatistics == null || nodeSummaryStatistics.size() != 2 ? false : true;
+        }, 30, TimeUnit.SECONDS);
+
+        List<NodeSummaryStatistics> nodeSummaryStatisticsList = new ArrayList<>(
+                analyticsService.getNodeSummaryStatistics(null, requestCriteria()));
+        Assert.assertEquals(2, nodeSummaryStatisticsList.size());
+
+        NodeSummaryStatistics nodeSummaryStatistics = nodeSummaryStatisticsList.get(0);
+        Assert.assertEquals("/root", nodeSummaryStatistics.getUri());
+        Assert.assertEquals("consumer", nodeSummaryStatistics.getComponentType());
+
+        nodeSummaryStatistics = nodeSummaryStatisticsList.get(1);
+        Assert.assertEquals("/serviceA", nodeSummaryStatistics.getUri());
+//        TODO this is sometimes producer and sometimes consumer? (if the wait interval is longer it is consumer)
```

Comment:
I think this is because one node summary stats is being overwritten by another - there should be three nodes.